File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedintegration/addremove/AddRemoveIntegrationActivity.java
package com.firewok.example.java_integration.feedintegration.addremove;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.widget.FrameLayout;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import com.firework.common.feed.FeedResource;
import com.firewok.example.java_integration.R;
import com.firewok.example.java_integration.databinding.ActivityAddRemoveIntegrationBinding;
import com.firework.videofeed.FwVideoFeedView;
import com.firework.viewoptions.BaseOption;
import com.firework.viewoptions.ViewOptions;

public class AddRemoveIntegrationActivity extends AppCompatActivity {
    private ActivityAddRemoveIntegrationBinding binding;
    private FwVideoFeedView videoFeedView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityAddRemoveIntegrationBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());
        
        setTitle(R.string.add_remove_screen_title);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.add_remove_screen_title));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
        }
        
        initAddRemoveButtons();
    }

    private void initAddRemoveButtons() {
        binding.addFeed.setOnClickListener(v -> {
            addFeed();
            updateButtonsAvailability();
        });

        binding.removeFeed.setOnClickListener(v -> {
            removeFeed();
            updateButtonsAvailability();
        });

        updateButtonsAvailability();
    }

    private void addFeed() {
        FwVideoFeedView videoFeedView = new FwVideoFeedView(this);

        ViewOptions viewOptions = new ViewOptions.Builder().baseOption(new BaseOption.Builder().feedResource(FeedResource.Discovery.INSTANCE).build()).build();


        videoFeedView.init(viewOptions);

        FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.MATCH_PARENT,
                FrameLayout.LayoutParams.MATCH_PARENT
        );
        binding.feedContainer.addView(videoFeedView, layoutParams);

        this.videoFeedView = videoFeedView;
    }

    private void removeFeed() {
        if (videoFeedView != null) {
            binding.feedContainer.removeView(videoFeedView);
            videoFeedView.destroy();
            this.videoFeedView = null;
        }
    }

    private void updateButtonsAvailability() {
        boolean isFeedAdded = this.videoFeedView != null;
        binding.addFeed.setEnabled(!isFeedAdded);
        binding.removeFeed.setEnabled(isFeedAdded);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        if (videoFeedView != null) {
            videoFeedView.destroy();
        }
        super.onDestroy();
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, AddRemoveIntegrationActivity.class);
    }
} 

File: feed_integration/app/src/main/java/com/firework/example/feedintegration/addremove/AddRemoveIntegrationActivity.kt
package com.firework.example.feedintegration.addremove

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import com.firework.common.feed.FeedResource
import com.firework.example.feedintegration.R
import com.firework.example.feedintegration.databinding.ActivityAddRemoveIntegrationBinding
import com.firework.videofeed.FwVideoFeedView
import com.firework.viewoptions.baseOptions
import com.firework.viewoptions.viewOptions

class AddRemoveIntegrationActivity : AppCompatActivity() {

    private lateinit var binding: ActivityAddRemoveIntegrationBinding

    private var videoFeedView: FwVideoFeedView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAddRemoveIntegrationBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        setTitle(R.string.add_remove_screen_title)
        supportActionBar?.title = getString(R.string.add_remove_screen_title)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)
        initAddRemoveButtons()
    }

    private fun initAddRemoveButtons() {
        binding.addFeed.setOnClickListener {
            addFeed()
            updateButtonsAvailability()
        }

        binding.removeFeed.setOnClickListener {
            removeFeed()
            updateButtonsAvailability()
        }

        updateButtonsAvailability()
    }

    private fun addFeed() {
        val videoFeedView = FwVideoFeedView(this)

        val viewOptions = viewOptions {
            baseOptions {
                feedResource(FeedResource.Discovery)
            }
        }

        videoFeedView.init(viewOptions)

        binding.feedContainer.addView(videoFeedView, FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT)

        this.videoFeedView = videoFeedView
    }

    private fun removeFeed() {
        val videoFeedView = requireNotNull(this.videoFeedView)

        binding.feedContainer.removeView(videoFeedView)

        videoFeedView.destroy()

        this.videoFeedView = null
    }

    private fun updateButtonsAvailability() {
        val isFeedAdded = this.videoFeedView != null

        binding.addFeed.isEnabled = !isFeedAdded
        binding.removeFeed.isEnabled = isFeedAdded
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onDestroy() {
        videoFeedView?.destroy()
        super.onDestroy()
    }

    companion object {
        fun intent(context: Context) = Intent(context, AddRemoveIntegrationActivity::class.java)
    }
}
